packages/service-integration/src/service_integration/osparc_image_specs.py

```diff
@@ -26,7 +28,7 @@ def create_image_spec(
 
     - the image-spec simplifies building an image to ``docker compose build``
     """
-    labels = {**meta_cfg.to_labels_annotations()}
+    labels: dict[str, str | float | bool | None] = {**meta_cfg.to_labels_annotations()}
```

Comment:
you know that you are effectively casting whatever comes back from to_labels_annotations here? that does not mean that is what you get. is that function not typed correctly?

Reply:
very good point, I have unified the typing on all functions.